File: AngularShopBackend/Src/Domain/Entities/Order/Order.cs
using Domain.Entities.Base;
using Domain.Entities.Identity;
using Domain.Enums;

namespace Domain.Entities.Order;

public class Order : BaseAuditableEntity
{

    public string BuyerPhoneNumber { get; set; }
    public string InvoiceNumber { get; set; }
    public int SubTotal { get; set; }
    public DateTime? PaymentDate { get; set; }
    public string Authority { get; set; }
    public string TrackingCode { get; set; }
    public long TransactionId { get; set; }
    public string BasketId { get; set; } = string.Empty;

    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public PortalType PortalType { get; set; } = PortalType.Novino;
    public bool IsFinally { get; set; } = false; 


    public List<OrderItem> OrderItems { get; set; } = new();
    public ShipToAddress ShipToAddress { get; set; }
    public DeliveryMethod DeliveryMethod { get; set; }
    public User User { get; set; }
}

File: AngularShopFrontend/src/app/featuers/profile/models/order.ts
import { PortalTypeEnum } from "../../checkout/models/checkout";
import { ShipToAddress } from "./address";


export interface Order {
  id: number;
  paymentDate: string;
  invoiceNumber: string;
  buyerPhoneNumber: string;
  subTotal: number;
  isFinally: boolean;
  transactionId: number;
  trackingCode: string;
  authority: string;
  portalType: number | string | PortalTypeEnum;
  orderStatus: number | string | OrderStatusEnum;
  deliveryMethod: DeliveryMethod;
  shipToAddress: ShipToAddress;
  orderItems: OrderItem[];
}

export interface DeliveryMethod {
  id: number;
  isDelete: boolean;
  shortName: string;
  deliveryTime: string;
  description: string;
  price: number;
}

export interface OrderRequest {
  basketId?: string;
  deliveryMethodId?: number;
  buyerPhoneNumber?: string;
  portalType?: number;
  shipToAddress?: ShipToAddress;
}

export enum OrderStatusEnum {
  درحال_بررسی = 1,
  درحال_پردازش,
  تحویل_اداره_پست,
  ارسال_شده,
  تحویل_داده_شده,
  بازگشت_داده_شده,
  انصراف_داده_شده,
  ناموفق
}

export interface OrderItem {
  id: number;
  price: number;
  quantity: number;
  itemOrdered: ItemOrdered;
}

export interface ItemOrdered {
  productItemId: number;
  productName: string;
  productTypeName: string;
  productBrandName: string;
  pictureUrl: string;
}
